be2-scala/src/main/scala/ch/epfl/pop/decentralized/HeartbeatGenerator.scala

```diff
@@ -0,0 +1,70 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.{Actor, ActorLogging, Props}
+import akka.pattern.AskableActorRef
+import ch.epfl.pop.model.network.method.Heartbeat
+import ch.epfl.pop.model.objects.{Channel, DbActorNAckException, Hash}
+import ch.epfl.pop.pubsub.AskPatternConstants
+import ch.epfl.pop.pubsub.graph.ErrorCodes
+import ch.epfl.pop.storage.DbActor
+import scala.util.{Failure, Success}
+import scala.collection.immutable.HashMap
+import scala.concurrent.Await
+
+// This actor will create heartbeat when asked to and send them to the actorRef it received
+case class HeartbeatGenerator(dbRef: AskableActorRef) extends Actor with ActorLogging with AskPatternConstants() {
+
+  /** retrieves the channels and their associated message ids
+    * @return
+    *   a map from the channels to message ids, None when errors are encountered
+    */
+  private def retrieveHeartbeatContent(): Option[HashMap[Channel, Set[Hash]]] = {
+    val askForChannels = dbRef ? DbActor.GetAllChannels()
+    val setOfChannels: Set[Channel] = Await.ready(askForChannels, duration).value match {
+      case Some(Success(DbActor.DbActorGetAllChannelsAck(set))) => set
```

Comment:
```suggestion
      case Some(Success(DbActor.DbActorGetAllChannelsAck(set))) => 
      set
```
Here too, maybe skipping a line makes it a bit more readable and consistent.